pycheribuild/qemu_utils.py

```diff
@@ -89,13 +89,20 @@ def __init__(self, xtarget: CrossCompileTarget, want_debugger=False):
             raise ValueError("Unknown target " + str(xtarget))
 
     def disk_image_args(self, image) -> list:
+        # Probe to see if this is qcow2 or not.  If it isn't, we assume it's raw, which isn't exercising the full range
+        # of qemu's features but we haven't, as yet, generated anything else.  We don't let qemu probe because it warns
+        # loudly if probing detects a raw image.
+        with open(image, 'rb') as imgf:
```

Comment:
I think probing is still useful even with #237. --pretend can easily be fixed by checking for .exists().

Reply:
i.e. something like:
```python
if self.config.pretend and not image.exists():
     img_fmt = "qcow2" if image.name.endswith(".qcow2") else "raw"
else:
    with image.open('rb') as imgf:
        magic = imgf.read(4)
        img_fmt = "qcow2" if magic == b'QFI\xfb' else "raw"
```

Reply:
Instead of the format parameter or if the parameter is `None` or to cross-check the parameter or...?

Reply:
I was thinking this could beb useful if someone explicitly overwrites the disk image path on the command line, when using the defaults the format argument should be sufficient.

Reading 4 bytes as a sanity check unconditionally should be fine for the default case if that makes the logic simpler?